### ISSUE TITLE
Automated Key Rotation Not Occurring Every 30 Days

### ISSUE THREAD

**Issue by ChefAustin** [hashicorp/nomad#23528]
### Nomad version
```
$ nomad version
Nomad v1.8.0+ent
BuildDate 2024-05-28T18:08:23Z
Revision 5a7a5d80944f1568c8bcd4e71a7f923fc04f2461
```

### Operating system and Environment details
```
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.4 LTS"
```

### Issue
[Possibly related to #19669 ? `¯\_(ツ)_/¯`]

According to the [‘Key Rotation’ section of Hashicorp Nomad's Key Management Operation Docs](https://developer.hashicorp.com/nomad/docs/operations/key-management#key-rotation), _“Nomad automatically rotates the active encryption key every 30 days”._

```
$ nomad operator root keyring list -verbose
Key                                   State   Create Time
c5b22e0f-0fdf-5ae6-6b81-8bec90df57c9  active  2023-08-03T08:23:36+02:00
```
Note how on my cluster, the key's `Create Time` is greater than 30 days (indicating that, perhaps, the recurring 30 day task for key rotation is not occurring as it should be).

Note: It is entirely possible that I am misunderstanding some nuances of how the keyring and key rotation work.

(Also: This issue dupes internal JIRA ticket NET-10398 && Hashicorp Support request #154558.)

### Reproduction steps
- Bring online a Nomad cluster using v1.6.1
- Upgrade to v1.7.3
- Upgrade to v1.7.3-ent
- Upgrade to v1.7.4-ent
- Create a second cluster (v1.7.4-ent) in a different region and federate it with the first cluster
- Upgrade to v1.8.0-ent
- Wait 31 days
- Check keyring list for active key's "Create Time"

#### Expected Result
Active key's `Create Time` should not be more than 30 days old.

#### Actual Result
Active key's `Create Time` is older than 30 days (342 days old in my case).

**Comment by tgross** [comment]
Hi @ChefAustin! I don't think federation is really involved here, as we don't replicate the keyring between regions (see https://github.com/hashicorp/nomad/issues/20123 for more discussion of that). But there's obviously something broken here for sure. I'm actively working on https://github.com/hashicorp/nomad/issues/19669 (and a host of other keyring-related issues), so I'll investigate this as part of that body of work.

(Also, thanks for the pointer to [NET-10398](https://hashicorp.atlassian.net/browse/NET-10398); it hadn't shown up on the Nomad engineering board yet.)

**Comment by tgross** [comment]
Hi @ChefAustin just a heads up that I've figured out this is happening because of https://github.com/hashicorp/nomad/issues/16359. I'm not going to block fixing this on solving #16359 more generally though; we have the `CreateTime` on the key metadata already so I can use that instead of the `CreateIndex` to check the rotation threshold.

**Comment by ChefAustin** [comment]
> Hi @ChefAustin just a heads up that I've figured out this is happening because of #16359. I'm not going to block fixing this on solving #16359 more generally though; we have the `CreateTime` on the key metadata already so I can use that instead of the `CreateIndex` to check the rotation threshold.

Great to hear that this bugfix isn't blocked by a larger effort. I appreciate your digging into this! Thanks.

**Comment by tgross** [comment]
Fixed in #23577 and will ship in the next regular release of Nomad 1.8.x, with backports to Nomad 1.7.x/1.6.x Enterprise.
